### ISSUE TITLE
[HOLD for payment 2023-08-16] [$1000] Web - The state name in the company information does not update to "Incorporation State," causing both state names to appear the same

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23870]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings > Workspaces
2. Select a workspace and access "Connect Bank Account."
3. Proceed to Step 2 out of 5.

## Expected Result:
The second state name in the company information should be changed to "Incorporation State."

## Actual Result:
The state name in the company information does not update to "Incorporation State," causing both state names to appear the same

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.47-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/1296b51f-99b5-411a-a1b1-a87f69728201

https://github.com/Expensify/App/assets/93399543/3dedab47-6e41-4d7c-9ef3-29fd616bbd52

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690576478988209

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0180f3b211410f7a0c</li>
        <li>Upwork Job ID: 1686400237560549376</li>
        <li>Last Price Increase: 2023-08-10</li>
<li>Automatic offers: </li>
<ul>
<li>dhairyasenjaliya | Contributor | 25889382</li>
</ul></li>
    </ul>
</details>

**Comment by kevinksullivan** [comment]
@mananjadhav are you able to review the proposal above?

**Comment by mananjadhav** [comment]
Thanks for the bump. @dhairyasenjaliya's [proposal](https://github.com/Expensify/App/issues/23870#issuecomment-1656826034) looks good to me.

🎀 👀 🎀  C+ reviewed.

**Comment by francoisl** [comment]
Makes sense to me!

**Comment by mananjadhav** [comment]
@francoisl I can see the same [report](https://expensify.slack.com/archives/C049HHMV9SM/p1691034324165019) for `Personal Information` too. Should we handle this in a separate issue?

**Comment by mananjadhav** [comment]
Ohh thanks for clarifying. I did mark the report as duplicate, but then I read Personal Information, so I thought we missed something.

**Comment by mananjadhav** [comment]
Okay I just saw you've commented. thanks for that too 😄

**Comment by kevinksullivan** [comment]
@adelekennedy I am going OOO for a week, so tapping you in to help out in the interim. Thanks!

**Comment by mananjadhav** [comment]
@adelekennedy @kevinksullivan This was deployed to production, but `[Hold for Payment]` isn't updated in the title.

**Comment by mananjadhav** [comment]
I'll finish the checklist today. I think it is related to when we migrated to `StatePicker` component. Meanwhile @adelekennedy can you post the payment summary here?

**Comment by mananjadhav** [comment]
As I mentioned we added refactored the `StatePicker` component in this [PR](https://github.com/Expensify/App/pull/21049/files), but we didn't add the `label` property.

I've added a comment in the [PR](https://github.com/Expensify/App/pull/21049/files#r1295771691). I think the following point  covers the impacted components, but we missed testing it.

```
I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like Avatar, I verified the components using Avatar are working as expected)
```

I don't think we need a regression test here.

cc - @adelekennedy

**Comment by adelekennedy** [comment]
### Payouts due:

Issue Reporter: $250 @ayazhussain79
Contributor: $1000 @dhairyasenjaliya Upwork
Contributor+: $1000 @mananjadhav NewDot

Eligible for 50% #urgency bonus? **Y**

Upwork job is [here](https://www.upwork.com/jobs/~0180f3b211410f7a0c).

**Comment by JmillsExpensify** [comment]
Reviewed the details for @mananjadhav. This $1,5000 payment is approved in NewDot based on the BZ summary above.

**Comment by mananjadhav** [comment]
@JmillsExpensify Can you correct the amount from $15000 to  $1500. 🤑

**Comment by kevinksullivan** [comment]
stepping back in to handle final payment. @ayazhussain79 just sent an offer for reporting. Let me know when you accept!

**Comment by ayazhussain79** [comment]
@kevinksullivan offer accepted, Thank you

**Comment by adelekennedy** [comment]
@kevinksullivan everyone was already paid!

**Comment by ayazhussain79** [comment]
I think not everyone

**Comment by kevinksullivan** [comment]
yeah, @adelekennedy @ayazhussain79 didn't appear to get paid for reporting so I just sent them an offer

**Comment by kevinksullivan** [comment]
Alright, all set on payment. So what checklist are we waiting on now @adelekennedy ?
